misk/web/@misk/components/tslint.json

```diff
@@ -19,6 +19,7 @@
     "ordered-imports": [true],
     "no-console": [false],
     // "no-var-requires": true,
+    "no-implicit-dependencies": false,
```

Comment:
this is curious!

Reply:
Bit of a design decision to loosen the linting requirements to allow for a simpler `package.json`.

With it set to true, any `import x from 'libraryX'` must have `libraryX` declared explicitly as a dependency in the `package.json`.

One of the goals with `@misk/common` was to allow developers to import only that single dependency and then get access to all common libraries (React, BlueprintJS...) that are used across the tabs. A single dependency would also give better control over consistency of version numbers which would prevent cases where a tab developer bumps their React version that includes some breaking changes and then doesn't work interoperably with the rest of the app.

With a single dependency, two courses of action for imports are available to us. 

1) `@misk/common` imports and then re-exports all methods from all of the common libraries. Import statements then are `import React from '@misk/common'`. This would eliminate the "no-implicit-dependencies" error in linting.

2) `@misk/common` only has the common libraries set as dependencies so they will always be available to tab developers but won't be explicitly stated in `package.json` but their import statements will look identical to all online documentation for those libraries: `import React from 'react'`.

The current approach was to go with 2), use `@misk/common` as a single dependency to maintain version consistency across the tabs and reduce the need for tab developers to worry about managing their `package.json` versions.

Thoughts?

Reply:
Got it. Yeah, lets make the dependency manager do the dependency manager’s job.